### PR TITLE
SCD-1, SCD-2, SCD-3

### PR DESCRIPTION
**Pull request by Denis-Davidoff**

### CODE REVIEW

File: contracts/RestrictedToken.sol [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 pragma solidity 0.8.4;
 
-import "./ITransferRules.sol";
+import "./interfaces/ITransferRules.sol";
```
Comment (voyager-gold) [inline contracts/RestrictedToken.sol:5]: Nice catch

**Review by voyager-gold — DISMISSED** [review]
I like it moving `ITransferRules.sol` under `./interfaces` folder.
I saw it uses Truffle for deployment. We prefer Hardhat over Truffle. Can you please spare some time to transit to Hardhat?
You may refer to https://github.com/CoMakery/comakery-security-token/pull/41

File: contracts/RestrictedToken.sol (hunk 1 above)
Comment (Denis-Davidoff) [reply contracts/RestrictedToken.sol:5]: Thanks)
I will start transferring to hardhat soon

**Comment by voyager-gold** [comment]
A funny thing is we have different size of compiled byte code between hardhat and truffle though the solc version and evm version is the same.

**Comment by voyager-gold** [comment]
@Denis-Davidoff 
Sorry, you don't have to work on this. There was a misconfiguration on my branch. Thanks.

**Comment by Denis-Davidoff** [comment]
> A funny thing is we have different size of compiled byte code between hardhat and truffle though the solc version and evm version is the same.

Yes, I agree, this is strange

**Comment by Denis-Davidoff** [comment]
I am assigned tasks in JIRA, it is not me who decides what to do.

**Comment by aquabu** [comment]
@Denis-Davidoff It looks like @voyager-gold made progress on the contract size issue and deployment on hardhat in this PR: https://github.com/CoMakery/comakery-security-token/pull/41

@voyager-gold is going to clarify how to deploy to different networks and then we can merge that PR.

To make this PR easier to merge, can you remove the truffle and deployment related code. It helped for debugging but I think it's going to make it hard to merge both PRs.

**Comment by Denis-Davidoff** [comment]
> @Denis-Davidoff It looks like @voyager-gold made progress on the contract size issue and deployment on hardhat in this PR: #41
> 
> @voyager-gold is going to clarify how to deploy to different networks and then we can merge that PR.
> 
> To make this PR easier to merge, can you remove the truffle and deployment related code from this PR #47 . It helped for debugging but I think it's going to make it hard to merge both PRs.

Maybe better merge brachs before and then i can remove truffle. This way there will be fewer dependencies.

**Comment by aquabu** [comment]
@Denis-Davidoff I merged PR #41 you can make modifications in this PR to remove the truffle configuration. Then I can merge this to master. Feel free to merge master into this branch first if it's helpful.

**Review by aquabu — APPROVED** [review]

**Pull request merged**